packages/xo-web/src/common/xo/index.js

```diff
@@ -564,6 +564,30 @@ export const createSrUnhealthyVdiChainsLengthSubscription = sr => {
 
 export const subscribeUserAuthTokens = createSubscription(() => _call('user.getAuthenticationTokens'))
 
+export const subscribeXoTasks = createSubscription(async previousTasks => {
+  let filter = ''
+  // Deduplicate previous tasks and new tasks with a Map
+  const tasks = new Map()
+  if (previousTasks !== undefined) {
+    let lastUpdate = 0
+    previousTasks.forEach(task => {
+      if (task.updatedAt > lastUpdate) {
+        lastUpdate = task.updatedAt
+      }
+      tasks.set(task.id, task)
+    })
+    filter = `&filter=updatedAt%3A%3E${lastUpdate}`
+  }
+
+  // Fetch new and updated tasks
+  ;(await fetch('/rest/v0/tasks?fields=*' + filter)).json().forEach(task => {
+    tasks.set(task.id, task)
+  })
```

Comment:
- best practice is to explicit the wanted fields (`*` is intended for easy exploration during dev/debug, and may be removed in the future)
- use relative instead of absolute paths to avoid breaking reverse proxies
- missing `await` for `.json()`
- use `for ()` instead of `.forEach()` to avoid ugly `;(` syntax

```suggestion
  const response = await fetch('./rest/v0/tasks?fields=end,id,name,objectId,properties,start,status,updatedAt' + filter)
  for (const task of await response.json()) {
    tasks.set(task.id, task)
  }
```